### PR TITLE
docs: add getting started guide for Swift

### PR DESCRIPTION
**Pull request by grdsdev**
## What kind of change does this PR introduce?

New docs for getting started guide for Swift library.

Preview: https://docs-fyujqbqzy-supabase.vercel.app/docs/guides/getting-started/tutorials/with-swift

### CODE REVIEW

File: apps/docs/pages/guides/auth/native-mobile-deep-linking.mdx [hunk 1]
```diff
@@ -306,6 +306,44 @@ In certain auth scenarios you will need to handle linking back into your applica
       </TabPanel>
     </Tabs>
   </TabPanel>
+
+  <TabPanel id="swift" label="Swift">
+  ### Deep link config
+
+    - Go to your [auth settings](https://supabase.com/dashboard/project/_/auth/url-configuration) page.
```
Comment (charislam) [inline apps/docs/pages/guides/auth/native-mobile-deep-linking.mdx:313]:
```suggestion
    1. Go to your [auth settings](https://supabase.com/dashboard/project/_/auth/url-configuration) page.
```

prefer ordered list for steps that must be performed in order

File: apps/docs/pages/guides/auth/native-mobile-deep-linking.mdx [hunk 2]
```diff
@@ -306,6 +306,44 @@ In certain auth scenarios you will need to handle linking back into your applica
...
+    - You need to enter your app redirect callback on `Additional Redirect URLs` field.
```
Comment (charislam) [inline apps/docs/pages/guides/auth/native-mobile-deep-linking.mdx:314]:
```suggestion
    2. Enter your app redirect URL in the `Additional Redirect URLs` field. This is the URL that the user gets redirected to after clicking a magic link.
```

File: apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+
+## Bonus: Profile photos
+
+Every Supabase project is configured with [Storage](/docs/guides/storage) for managing large files like
+photos and videos.
+
+### Add PhotosPicker
+
+Let's add support for the user to pick an image from the library and upload it.
+We can start by creating a new type to hold the picked avatar image:
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx:297]:
```suggestion
Let's add support for the user to pick an image from the library and upload it.
Start by creating a new type to hold the picked avatar image:
```

File: apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+    self.init(image: image, data: data)
+  }
+}
+
+enum TransferError: Error {
+  case importFailed
+}
+```
+
+#### Add PhotosPicket to Profile page
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx:333]:
```suggestion
#### Add PhotosPicker to Profile page
```

File: apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
++        file: data,
++        options: FileOptions(contentType: "image/jpeg")
++      )
++
++    return filePath
++  }
+}
+```
+
+And finally let's update our Models.
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx:499]:
```suggestion
Finally, update your Models.
```

File: apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+  enum CodingKeys: String, CodingKey {
+    case username
+    case fullName = "full_name"
+    case website
+    case avatarURL = "avatar_url"
+  }
+}
+```
+
+We no longer need the `UpdateProfileParams` struct, as we can now reuse the `Profile` struct for both request and response calls.
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-swift.mdx:517]:
```suggestion
You no longer need the `UpdateProfileParams` struct, as you can now reuse the `Profile` struct for both request and response calls.
```

**Review by charislam — APPROVED** [review]
Looking good, thanks! 🎉 

Some minor nits for concision and style + you'll need to add this to the nav

**Review by thorwebdev — APPROVED** [review]
Awesome, thanks @grdsdev. The prettier check is failing. In the root of `supabase` repo could you please run `npm run format` and commit any changes that this produced?

**Comment by thorwebdev** [comment]
Ah, and could you include the example app code here: https://github.com/supabase/supabase/tree/master/examples/user-management ?

**Comment by grdsdev** [comment]
Done @thorwebdev can you approve again? Thank you!

**Pull request merged**